sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -183,9 +183,15 @@ case class CatalogTable(
 
   import CatalogTable._
 
-  /** schema of this table's partition columns */
-  def partitionSchema: StructType = StructType(schema.filter {
-    c => partitionColumnNames.contains(c.name)
+  /**
+   * schema of this table's partition columns
+   * keep the schema order with partitionColumnNames
```

Comment:
"keep the schema order with partitionColumnNames because we always concatenate the partition columns to the schema when reading the table information from hive  metastore."